### ISSUE TITLE
Unexpected parent state and chaining warnings

### ISSUE THREAD

**Issue by Jazzmanpw** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#18]
Hi again! I have a hook with a confirmation modal (e.g. _Are you sure you want to delete this item?_). There are cases, though, when confirmation is not needed, and the final action should be performed immediately, and I handle it in an effect (because in case of a skipped confirmation I need to set some states, and I can't do that during render, but I also can't set it as the default value). Here's simplified version of the hook and errors I get.

```tsx
export default function useConfirmation<T>({
  shouldSkipConfirmation,
  onConfirm,
}: UseConfirmationOptions<T>) {
  const [data, setData] = useState<T | null>(null);
  const [skipsConfirmation, setSkipsConfirmation] = useState(false);

  useEffect(() => {
    if (data !== null && !skipsConfirmation && shouldSkipConfirmation?.(data)) {
      //                                       ^ Avoid coupling parent behavior or state to a child component. Instead, lift shared logic or state up to the parent.
      setSkipsConfirmation(true);
      // ^ Avoid chaining state changes. When possible, update all relevant state simultaneously.
      onConfirm?.(data, () => {
      // ^ Avoid coupling parent behavior or state to a child component. Instead, lift shared logic or state up to the parent.
        close();
        setSkipsConfirmation(false);
      });
    }
  }, [data, onConfirm, shouldSkipConfirmation, skipsConfirmation]);

  const component = data !== null && !skipsConfirmation && (
    <SomeComponents>It doesn't matter</SomeComponents>
  );

  return {
    child: component,
    show: (state: T) => setData(state),
  };
}
```

The usage is roughly like this (here, you don't need to confirm deletion of your own items), so it's nothing about parent state

```tsx
const myConfirmation = useConfrimation<{id: number, isMyItem: boolean}>({
  shouldSkipConfirmation: (data) => data.isMyItem,
  async onConfirm(data, onClose) {
    await deleteStuffAsync(data);
    onClose();
  },
});

return (
  <ul>
    <li>
      <span>My item</span>
      <button onClick={() => myConfirmation.show({id: 1, isMyItem: true})}>
        Delete
      </button>
    </li>
    <li>
      <span>Their item</span>
      <button onClick={() => myConfirmation.show({id: 2, isMyItem: false})}>
        Delete
      </button>
    </li>
  </ul>
);
```

I'm not sure if that's me doing something wrong or if it's a false positive warning from the plugin. I disable the errors for now, but will be happy to discuss.

The error appeared when I upgraded the package from `0.0.38` to `0.1.5` (I can try to find a more exact version where it broke, if it helps)
